sycl/test/plugins/sycl-targets-order.cpp

```diff
@@ -0,0 +1,70 @@
+// RUN: %clangxx -fsycl -fsycl-targets=spir64-unknown-unknown-sycldevice,nvptx64-unknown-unknown-sycldevice %s -o %t-spir64-nvptx64.out
+// RUN: env SYCL_BE=PI_OPENCL %t-spir64-nvptx64.out
+// RUN: env SYCL_BE=PI_CUDA   %t-spir64-nvptx64.out
+// RUN: %clangxx -fsycl -fsycl-targets=nvptx64-unknown-unknown-sycldevice,spir64-unknown-unknown-sycldevice %s -o %t-nvptx64-spir64.out
+// RUN: env SYCL_BE=PI_OPENCL %t-nvptx64-spir64.out
+// RUN: env SYCL_BE=PI_CUDA   %t-nvptx64-spir64.out
+
+// REQUIRES: opencl, cuda
+
+//==------- sycl-targets-order.cpp - SYCL -fsycl-targets order test --------==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include <CL/sycl.hpp>
+#include <iostream>
+
+int main(int argc, char **argv) {
+
+  // select the default SYCL device
+  cl::sycl::device device{cl::sycl::default_selector{}};
```

Comment:
Default selector can select default device so the test will be silently passing while checking nothing.
I suggest either writing own selector which skips host device and if host device is the only device available  fail the test.
Or add %GPU_RUN_PLACEHOLDER in the RUN line(see other tests for examples)

Reply:
The device does not need to be a GPU.

Reply:
Then, please, implement selector which makes sure host device is not chosen.

Reply:
Or at lest put check that selected device is not host one, so we can detect it in future.

Reply:
In fact, if there are no OpenCL devices or no CUDA devices the test will already fail:
```
OCL_ICD_VENDORS= SYCL_BE=PI_OPENCL ./sycl-targets-order-spir64-nvptx64.out

terminate called after throwing an instance of 'cl::sycl::runtime_error'
  what():  No device of requested type available. 0 (CL_SUCCESS)
Aborted (core dumped)
```

Reply:
Ok, but this behavior seems like a spec violation to me.

Reply:
`SYCL_BE` and the plugin mechanism are not covered by the specs, AFAIK.

I understand the effect of `SYCL_BE` as forcing the default selector to use a specific platform (or set of platforms); with this interpretation it makes sense to me that it prevents using the host device.